cpp/src/arrow/util/decimal.h

```diff
@@ -104,17 +104,15 @@ class ARROW_EXPORT Decimal128 {
   /// \brief Get the low bits of the two's complement representation of the number.
   uint64_t low_bits() const { return low_bits_; }
 
-  /// \brief Put the raw bytes of the value into a pointer to uint8_t.
-  Status ToBytes(std::array<uint8_t, 16>* out) const;
+  /// \brief Return the raw bytes of the value.
+  std::array<uint8_t, 16> ToBytes() const;
```

Comment:
It might be worth having a void ToBytes(uint8_t* out) const version, too, for writing into preallocated memory